dali/python/nvidia/dali/pipeline.py

```diff
@@ -1568,6 +1570,7 @@ def checkpoint(self, filename=None):
         external_ctx_cpt.pipeline_data = pickle.dumps(
             {"iter": self._consumer_iter, "epoch_idx": self._epoch_idx}
         )
+        external_ctx_cpt.iterator_data = self._iterator_data or ""
```

Comment:
It's a nitpick, but I'd rather make the iterator_data explicit.

It seems to me it would be easier to think of the `_iterator_data` the same way as about the `_checkpoint` - something static/readonly throught execution that describes the state we resumed from (and given instance will ever resume once). It's there seprately from _checkpoint as a necessary evil because it is the backend that can parse it out from the _checkpoint and, with the way most iterators are created, we don't want user to pass checkpoint twice.

But the fact that there is special private field in the pipeline to set from the iterator before calling the checkpoint method seems like one more unexpected twist in the (internal but still) API.

We could have `._checkpoint` and `checkpoint` methods with the later calling the first and the first accepting the iterator_data argument that is not present in the `checkpoint` method not to expose it in public API. It makes little difference to what's there now, but I think it's clearer how to use it properly (plus the pipeline._iterator_data does not evolve). It feels to me like less mental burden for the future developers. :)

Reply:
That's a great idea, thank you! I changed it in https://github.com/NVIDIA/DALI/pull/5371/commits/4be32487c0ce79790a5327dee27b065ffe950e65 , is it ok now?